lib/src/main/java/space/taran/arklib/domain/index/ResourceMeta.kt

```diff
@@ -1,14 +1,63 @@
 package space.taran.arklib.domain.index
 
+import space.taran.arklib.ResourceId
+import space.taran.arklib.computeId
+import space.taran.arklib.domain.dao.ResourceWithExtra
+import space.taran.arklib.domain.kind.GeneralKindFactory
+import space.taran.arklib.domain.kind.ResourceKind
+import space.taran.arklib.domain.meta.MetadataStorage
+import space.taran.arklib.utils.MetaResult
+import space.taran.arklib.utils.extension
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.attribute.FileTime
 
-typealias ResourceIdLegacy = Long
-
 data class ResourceMeta(
-    val id: ResourceIdLegacy,
+    val id: ResourceId,
     val name: String,
     val extension: String,
     val modified: FileTime,
     val size: Long,
```

Comment:
We can omit this field now and use `id.dataSize` instead. Tiny shortcut-function can be implemented though like `fun size() = id.dataSize`.